cmd/deck/static/configured-jobs/configured-jobs.ts

```diff
@@ -0,0 +1,166 @@
+            repoContainer.appendChild(createJobCard(job));
+        });
+    } else {
+        const message = document.createElement("h3");
+        message.innerHTML = "No Jobs found for " + repo.org.name + "/" + repo;
+        repoContainer.appendChild(message);
+    }
+}
+
+/**
+ * Creates and returns a card for the provided job
+ */
+function createJobCard(job: Job): HTMLElement {
+    const title = document.createElement("h3")
+    title.innerHTML = job.name;
+    title.classList.add("mdl-card__title-text");
+    const cardTitle = document.createElement("div");
+    cardTitle.classList.add("mdl-card__title");
+    cardTitle.appendChild(title);
+
+    const cardDesc = document.createElement("div");
+    cardDesc.innerHTML = job.type;
+    cardDesc.classList.add("mdl-card__supporting-text");
+
+    const cardAction = document.createElement("div");
+    const actionButton = document.createElement("a");
+    actionButton.innerHTML = "Details";
+    actionButton.classList.add(...["mdl-button", "mdl-button--colored", "mdl-js-button", "mdl-js-ripple-effect"]);
+    actionButton.addEventListener("click", () => {
+        const dialogElement = document.querySelector("dialog");
+        const titleElement = dialogElement.querySelector(".mdl-dialog__title")!;
+        titleElement.innerHTML = job.name;
+        const contentElement = dialogElement.querySelector(".mdl-dialog__content")!;
+
+        while (contentElement.firstChild) {
+            contentElement.removeChild(contentElement.firstChild);
+        }
+
+        const container = document.createElement("div");
+        const sectionTitle = document.createElement("h5");
+        const sectionBody = document.createElement("div");
+        contentElement.appendChild(container);
+
+        sectionBody.classList.add("dialog-section-body");
+        sectionBody.innerHTML = genJobDetails(job);
+        sectionTitle.classList.add("dialog-section-title");
+        sectionTitle.innerHTML = "Job Definition";
+
+        container.classList.add("dialog-section");
+        container.appendChild(sectionTitle);
+        container.appendChild(sectionBody);
+        PR.prettyPrint();
+        dialogElement.showModal();
+    });
+    cardAction.appendChild(actionButton);
+    cardAction.classList.add("mdl-card__actions", "mdl-card--border");
+
+    const card = document.createElement("div");
+    card.appendChild(cardTitle);
+    card.appendChild(cardDesc);
+    card.appendChild(cardAction);
+    card.classList.add("job-definition-card", "mdl-card", "mdl-shadow--2dp");
+
+    return card;
+}
+
+/**
+ * Creates and returns the inner content of the modal display for the provided job
+ */
+function genJobDetails(job: Job): string {
+    return `
+        <div>
+            <summary>Type: ${job.type}</summary>
+            <pre class="prettyprint"><code class="language-yaml job-definition">${job.yamlDefinition}</code></pre>
+            <a href="${job.jobHistoryLink}">Job History</a>
+        </div>
+    `;
+}
```

Comment:
This is bigger than the original but is safer and performs better:

```suggestion
function genJobDetails<T extends HTMLElement>(job: Job): T {
    const div = document.createElement('div');
    const summary = document.createElement('summary');
    const pre = document.createElement('pre');
    const code = document.createElement('code');
    const a = document.createElement('a');
    
     a.href = job.jobHistoryLink;
     a.appendChild(document.createTextNode('Job History'));
     pre.className = 'prettyprint';
     code.className = 'language-yaml job-definition';
     
    code.appendChild(document.createTextNode(job.yamlDefinition));
    pre.appendChild(code);
    summary.appendChild(document.createTextNode(`Type: ${job.type}`));
    div.appendChild(summary);
    div.appendChild(pre);
    div.appendChild(a);
    
    return div as T;
}
```

> *PS*: I am not sure if the declaration and return type are right, also probably there is a "typescript way" to do that, but this can be done in the future.